### ISSUE TITLE
broken p2p communication in protocol-feature-foundations

### ISSUE THREAD

**Issue by taokayan** [EOSIO/eos#6851]
in branch ```protocol-feature-foundations```, if a block is received from a peer, it will likely be broken at the time it is broadcast to another peer, resulting in the following exception:
```
warn  2019-02-28T09:41:19.503 thread-0  controller.cpp:1489           push_block           ] 0 exception: unspecified
unable to reconstruct public key from signature
    {}
    thread-1  elliptic_secp256k1.cpp:160 public_key
error 2019-02-28T09:41:19.503 thread-0  producer_plugin.cpp:320       on_incoming_block    ] 0 exception: unspecified
unable to reconstruct public key from signature
    {}
    thread-1  elliptic_secp256k1.cpp:160 public_key
rethrow
    {}
    thread-0  controller.cpp:1489 push_block
```

This is unrelated to the speculative or irreversible mode, and can be reproduce by the following setup:
[BP node] -> [ peer 1 ] -> [ peer 2 ]
and the exception message is on peer2.

However if peer 1 is running the old code (release/1.6.x), this issue will not exist.

**Comment by taokayan** [comment]
#6736

**Comment by arhag** [comment]
Resolved by #6856.
